src/domains/Lotto/index.js

```diff
@@ -1,43 +1,38 @@
-import {
-  MAX_LOTTO_NUMBER_COUNT,
-  MAX_LOTTO_NUMBER,
-  MIN_LOTTO_NUMBER,
-  ERROR_MESSAGES,
-} from "./constants/index.js";
+import { LottoNumber } from "../LottoNumber/index.js";
 
 export class Lotto {
-  #numbers;
+  static SIZE = 6;
+  static ErrorMessages = {
+    INVALID_LOTTO_NUMBERS: `로또는 ${Lotto.SIZE}개의 중복없는 숫자를 가진다`,
+  };
 
-  constructor(numbers) {
-    if (numbers.length !== MAX_LOTTO_NUMBER_COUNT) {
-      throw new Error(ERROR_MESSAGES.INVALID_COUNT);
-    }
-
-    if (new Set(numbers).size !== MAX_LOTTO_NUMBER_COUNT) {
-      throw new Error(ERROR_MESSAGES.DUPLICATE_NUMBERS);
-    }
+  #lottoNumbers;
 
-    if (
-      numbers.some(
-        (number) => number < MIN_LOTTO_NUMBER || number > MAX_LOTTO_NUMBER
-      )
-    ) {
-      throw new Error(ERROR_MESSAGES.OUT_OF_RANGE);
+  constructor(numbers) {
+    if (new Set(numbers).size !== Lotto.SIZE) {
+      throw new Error(Lotto.ErrorMessages.INVALID_LOTTO_NUMBERS);
     }
 
-    this.#numbers = numbers;
+    this.#lottoNumbers = new Set(
+      numbers.map((number) => new LottoNumber(number))
+    );
   }
 
   get numbers() {
-    return this.#numbers;
+    return Array.from(this.#lottoNumbers)
+      .map((lottoNumber) => lottoNumber.value)
+      .sort((a, b) => a - b);
+  }
+
+  countMatchingLottoNumber(lotto) {
+    return [...this.#lottoNumbers].filter((lottoNumber) =>
+      lotto.hasLottoNumber(lottoNumber)
+    ).length;
   }
```

Comment:
```suggestion
  matchCount(lotto) {
    return [...this.#lottoNumbers].filter((lottoNumber) => lotto.has(lottoNumber)).length;
  }
```

lottoNumber 라는 suffix가 쓰이고 있는데요, 없어도 표현하는데 무방해보여요! lotto라는 namespace가 있으니까 간단하게 표현해보면 어떨까요!?